### PR TITLE
[1161] Remove the `unconditional_offers_via_api` feature flag

### PR DESCRIPTION
**Pull request by avinhurry**
## Context

Removing the `unconditional_offers_via_api` feature flag as its been active in production since March 2021.

## Changes proposed in this pull request

See commits

## Link to Trello card

https://trello.com/c/Xj3L5cwh/1161-apply-remove-the-unconditionaloffersviaapi-feature-flag

## Things to check

- [x] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [x] API release notes have been updated if necessary
- [x] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Comment by avinhurry** [comment]
[Apply: Remove the unconditional_offers_via_api feature flag](https://trello.com/c/Xj3L5cwh/1161-apply-remove-the-unconditionaloffersviaapi-feature-flag)

File: app/services/data_migrations/remove_unconditional_offers_via_api_feature_flag.rb
Comment (JR-G) [inline app/services/data_migrations/remove_unconditional_offers_via_api_feature_flag.rb:1]: Looks good Avin! Could we split this into two PRs please – the migration should follow in a subsequent migration 🙏

Comment (avinhurry) [reply app/services/data_migrations/remove_unconditional_offers_via_api_feature_flag.rb:1]: Sure, no problem. 

What's the reason for this?

Comment (avinhurry) [reply app/services/data_migrations/remove_unconditional_offers_via_api_feature_flag.rb]: Ok fair point.

Moved the data migration to this PR: https://github.com/DFE-Digital/apply-for-teacher-training/pull/8987, will merge after this PR is merged and a rebase has been done.

**Review by JR-G — APPROVED** [review]

**Pull request merged**
